eyelid-match-ops/src/primitives/poly.rs

```diff
@@ -130,15 +132,31 @@ pub fn extended_gcd<const MAX_POLY_DEGREE: usize>(
 
 /// This sampling is similar to what will be necessary for YASHE KeyGen.
 /// The purpose is to obtain a polynomial with small random coefficients.
-pub fn sample<const MAX_POLY_DEGREE: usize>(mut rng: ThreadRng) -> Poly<MAX_POLY_DEGREE> {
+pub fn sample_gaussian<const MAX_POLY_DEGREE: usize>(mut rng: ThreadRng) -> Poly<MAX_POLY_DEGREE> {
     let mut res = Poly::zero();
     // TODO: assert that this is less than the modulus of the coefficient
-    let max_coeff = 6;
-    let t = 2;
     for i in 0..MAX_POLY_DEGREE {
-        let coeff_rand = rng.gen_range(1..max_coeff);
-        res[i] = Coeff::from(t * coeff_rand);
+        // TODO: use delta (YASHE param) instead of 3.2
+        // Then those sampling functions should be moved
+        let normal = Normal::new(0.0, 3.2).unwrap();
+        let v: f64 = normal.sample(&mut rng);
+        res[i] = Coeff::from(v as i64);
     }
     res[0] += Coeff::one();
+    res.truncate_to_canonical_form();
+    res
+}
+
+/// This sampling is similar to what will be necessary for YASHE KeyGen.
+/// The purpose is to obtain a polynomial with small random coefficients.
+pub fn sample_rand<const MAX_POLY_DEGREE: usize>(mut rng: ThreadRng) -> Poly<MAX_POLY_DEGREE> {
+    let mut res = Poly::zero();
+    // TODO: assert that this is less than the modulus of the coefficient
+    for i in 0..MAX_POLY_DEGREE {
+        // TODO: implement Coeff:rand
+        let coeff_rand = rand_coeff(&mut rng);
```

Comment:
If you import `ark_ff::UniformRand` into this module, you might be able to just do this:
```suggestion
        let coeff_rand = Coeff::rand(&mut rng);
```

Then there’s no need for any changes to `fq.rs`.

In general, we should be able to use `Coeff` everywhere, and never need to use any `Fq` types. Ticket #59 will make this easier.